### PR TITLE
[7392] fix conformance test in PHP JSON parser

### PR DESCRIPTION
**Pull request by antongrbin**
# Motivation

This PR fixes failing JSON conformance tests for php with name `IgnoreUnknownEnumStringValue*`.

The JSON parsing spec was discussed in https://github.com/protocolbuffers/protobuf/issues/7392.

Recent similar changes in other languages:
- Python: https://github.com/protocolbuffers/protobuf/commit/86abf35ef5ee5b1004ec11bebb36d84c2ef6645e
- Swift: https://github.com/apple/swift-protobuf/pull/1345
- C#: https://github.com/protocolbuffers/protobuf/pull/15758
- C++: https://github.com/protocolbuffers/protobuf/pull/16479

### CODE REVIEW

**Comment by antongrbin** [comment]
@jskeet, since you worked with me on the linked issue https://github.com/protocolbuffers/protobuf/issues/7392, is it OK to ask you to help me find a PHP reviewer to move this forward?

**Comment by bshaffer** [comment]
@antongrbin I can take a look

**Comment by antongrbin** [comment]
@bshaffer thank you for the review offer && polite ping :)

**Comment by antongrbin** [comment]
@bshaffer friendly ping :)

**Review by bshaffer — COMMENTED** [review]
This is really the behavior we want? If an array/map is unknown, it throws an exception unless _other_ valid values exist, at which point the unknown value is ignored? 

That does not seem like intuitive behavior to me... what is the justification for this?

**Review by bshaffer — CHANGES_REQUESTED** [review]

File: php/tests/EncodeDecodeTest.php [hunk 1]
```diff
@@ -205,6 +205,56 @@ public function testDecodeRepeatedStringValue()
...
+
+    public function testDecodeRepeatedEnumWithUnknownStringValueIgnored()
+    {
+        $m = new TestMessage();
+        $m->mergeFromJsonString(
+          "{\"repeated_enum\":[
+            \"ONE\",
+            \"UNKNOWN_ENUM\",
+            \"ZERO\"
+          ]}",
```
[the remark this answers is not in the document]
Comment (antongrbin) [reply php/tests/EncodeDecodeTest.php:249]: Great catch! I now made the JSON string inputs equal in each test pair.

**Comment by antongrbin** [comment]
@bshaffer thank you for your time for reviewing this and sorry for the delay.

I recently switched companies so I lost access to noom/protobuf and can't update this PR anymore.

I created a new equivalent PR here: https://github.com/protocolbuffers/protobuf/pull/19376.

**Pull request closed without merging**